src/deluge/gui/views/view.cpp

```diff
@@ -1176,6 +1176,10 @@ void View::modButtonAction(uint8_t whichButton, bool on) {
 		if (on) {
 
 			if (isUIModeWithinRange(modButtonUIModes) || (getRootUI() == &performanceSessionView)) {
+				// change the button selection before calling mod button action so that mod button action
+				// knows the mod button parameter context
+				*activeModControllableModelStack.modControllable->getModKnobMode() = whichButton;
+
 				activeModControllableModelStack.modControllable->modButtonAction(
 				    whichButton, true, (ParamManagerForTimeline*)activeModControllableModelStack.paramManager);
 
```

Comment:
The line below (1186) is duplicated and needs to be removed as it was moved above (1181)